source/Plugins.h

```diff
@@ -86,12 +106,21 @@ class Plugins {
 	// Returns true if any plugin enabled or disabled setting has changed since
 	// launched via user preferences.
 	static bool HasChanged();
+	// Returns true if there is active install/update activity;
+	static bool IsInBackground();
 
 	// Returns the list of plugins that have been identified by the game.
 	static const Set<Plugin> &Get();
 
 	// Toggles enabling or disabling a plugin for the next game restart.
 	static void TogglePlugin(const std::string &name);
+
+	// Install or update or delete a plugin.
+	static std::future<void> Install(InstallData *installData, bool update = false);
+	static std::future<void> Update(InstallData *installData);
+	static void DeletePlugin(const std::string &pluginName);
+
+	static bool Download(const std::string &url, const std::string &location);
```

Comment:
Maybe this should be moved to Files, as it's not doing anything specific to plugins.

Reply:
https://github.com/endless-sky/endless-sky/pull/9068#discussion_r1321846929
quyykk told me different